src/ScheduleFunctions.cpp

```diff
@@ -1412,30 +1412,114 @@ class InjectFunctionRealization : public IRMutator {
 
         user_assert(num_skipped == 0) << "Fused groups must either be entirely used or unused\n";
 
+        // Order of the stages for building produce definitions.
+        vector<pair<Function, int>> stage_order;
+        // Inverse map from function name to the index.
+        map<string, int> func_name_to_index;
+        // Contains a number of dependencies which need to go first for a given stage of the function.
+        vector<vector<int>> stage_dependencies(funcs.size());
+        // Adjacency list of dependencies.
+        vector<vector<vector<pair<int, int>>>> adj_list(funcs.size());
```

Comment:
Good point, I agree it wasn't very clear. I've replaced std::pair with the struct with named fields and expanded the comment to explain the structure. I thought about encapsulating all of it into the class, but it seems like it's going to be a wrapper around vector anyway, so that extra indirection may make it less readable. What do you think?